app/controllers/runtime/spaces_controller.rb

```diff
@@ -212,11 +211,14 @@ def delete_isolation_segment(guid)
 
     def before_update(obj)
       if request_attrs['isolation_segment_guid']
-        raise CloudController::Errors::ApiError.new_from_details('NotAuthorized') unless SecurityContext.admin?
+        check_isolation_segment_access!(obj)
 
-        isolation_segment = IsolationSegmentModel[guid: request_attrs['isolation_segment_guid']]
-        raise CloudController::Errors::ApiError.new_from_details('InvalidRelation',
-                "Could not find Isolation Segment with guid: #{request_attrs['isolation_segment_guid']}") unless isolation_segment
+        isolation_segment_guids = obj.organization.isolation_segment_models.map(&:guid)
```

Comment:
Sorry about the confusion. This was a note to myself that I forgot to edit.

I'm talking about `obj.organization.isolation_segment_models.map(&:guid)` vs something like `obj.organization.get_isolation_segment_guids`. That way you define a public interface for organizations instead of reaching into their internal state. We do the former a lot in CC anyway, so this isn't blocking the PR.